trace_decoder/src/type1.rs

```diff
@@ -1,27 +1,25 @@
 //! Frontend for the witness format emitted by e.g the [`0xPolygonZero/erigon`](https://github.com/0xPolygonZero/erigon)
 //! Ethereum node (a.k.a "jerigon").
 
-use std::{
-    array,
-    collections::{HashMap, HashSet},
-    iter,
-};
+use std::array;
+use std::collections::{BTreeMap, BTreeSet};
 
 use anyhow::{bail, ensure, Context as _};
 use either::Either;
-use mpt_trie::{
-    partial_trie::{HashedPartialTrie, PartialTrie as _},
-    trie_ops::ValOrHash,
-};
+use evm_arithmetization::generation::mpt::AccountRlp;
 use nunny::NonEmpty;
 use u4::U4;
 
+use crate::typed_mpt::{StateTrie, StorageTrie, TriePath};
 use crate::wire::{Instruction, SmtLeaf};
 
+#[derive(Debug, Default, Clone)]
 pub struct Frontend {
-    pub state: HashedPartialTrie,
-    pub code: HashSet<NonEmpty<Vec<u8>>>,
-    pub storage: HashMap<ethereum_types::H256, HashedPartialTrie>,
+    pub state: StateTrie,
+    pub code: BTreeSet<NonEmpty<Vec<u8>>>,
+    /// The key here matches the [`TriePath`] inside [`Self::state`] for
+    /// accounts which had inline storage
```

Comment:
```suggestion
    /// accounts which had inline storage.
```

Minor but we general try and aim for "good" grammar following `///`